plugins/common/adx/adx_test.go

```diff
@@ -0,0 +1,134 @@
+package adx
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io"
+	"testing"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	"github.com/Azure/azure-kusto-go/kusto/ingest"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestInitBlankEndpointData(t *testing.T) {
+	plugin := Config{
+		Endpoint: "",
+		Database: "mydb",
+	}
+
+	_, err := plugin.NewClient("TestKusto.Telegraf", nil)
+	require.Error(t, err)
+	require.Equal(t, "endpoint configuration cannot be empty", err.Error())
+}
+
+func TestQueryConstruction(t *testing.T) {
+	const tableName = "mytable"
+	const expectedCreate = `.create-merge table ['mytable'] (['fields']:dynamic, ['name']:string, ['tags']:dynamic, ['timestamp']:datetime);`
+	const expectedMapping = `` +
+		`.create-or-alter table ['mytable'] ingestion json mapping 'mytable_mapping' '[{"column":"fields", ` +
+		`"Properties":{"Path":"$[\'fields\']"}},{"column":"name", "Properties":{"Path":"$[\'name\']"}},{"column":"tags", ` +
+		`"Properties":{"Path":"$[\'tags\']"}},{"column":"timestamp", "Properties":{"Path":"$[\'timestamp\']"}}]'`
+	require.Equal(t, expectedCreate, createTableCommand(tableName).String())
+	require.Equal(t, expectedMapping, createTableMappingCommand(tableName).String())
+}
+
+type fakeIngestor struct {
+	actualOutputMetric []map[string]interface{}
+}
+
+func (f *fakeIngestor) FromReader(_ context.Context, reader io.Reader, _ ...ingest.FileOption) (*ingest.Result, error) {
+	scanner := bufio.NewScanner(reader)
+	scanner.Scan()
+	firstLine := scanner.Text()
+	err := json.Unmarshal([]byte(firstLine), &f.actualOutputMetric)
+	if err != nil {
+		return nil, err
+	}
+	return &ingest.Result{}, nil
+}
+
+func (*fakeIngestor) FromFile(_ context.Context, _ string, _ ...ingest.FileOption) (*ingest.Result, error) {
+	return &ingest.Result{}, nil
+}
+
+func (*fakeIngestor) Close() error {
+	return nil
+}
+
+func TestGetMetricIngestor(t *testing.T) {
+	plugin := Client{
+		logger: testutil.Logger{},
+		client: kusto.NewMockClient(),
+		cfg: &Config{
+			Database:      "mydb",
+			IngestionType: QueuedIngestion,
+		},
+	}
+
+	plugin.ingestors = map[string]ingest.Ingestor{
+		"test1": &fakeIngestor{},
+	}
+
+	ingestor, err := plugin.GetMetricIngestor(context.Background(), "test1")
+	if err != nil {
+		t.Errorf("Error getting ingestor: %v", err)
+	}
```

Comment:
```suggestion
	require.NoError(t, err)
```
Same for all other spots below!